lib/event.c

```diff
@@ -1690,6 +1690,35 @@ static void thread_process_io(struct event_loop *m, unsigned int num)
 	for (i = 0; i < last_read && ready < num; ++i)
 		thread_process_io_inner_loop(m, num, pfds, &i, &ready);
 
+	/*
+	 * At this point any fd's that were processed
```

Comment:
I'm a little confused about the issue: if a process closes an fd without cancelling the read or write task that uses the fd, that sounds like an application issue that should be fixed?

Reply:
it is cancelling the read/writes!  It's a different problem.  Here's the normal loop of events for read or writes:

program registers with the event system a read/write event.  ppoll fds array is modified appropriately
ppoll is run eventually.  Read or write event happens.  lib/event.c clears the read/write event leaving a .event with 0 and calls the appropriate handler. 
Handler wakes up and decides that no more reads or writes should be put on the ppoll fds and does not reregister itself.  <this leaves the fd on the ppoll fds array>.  
ppoll is entered again and the socket is closed and ppoll sends us a POLLERR which means that something bad happened with the fd, but since nothing got reregistered it puts out the warning.

Reply:
What I mean is that the array of fds - handler.pfds - isn't removing an fd after each event on the fd. doesn't it look like the fd is left in place, with zero poll bits set? what's clearing the fd from the pfds array, other than the event cancel apis - is it the POLLNVAL path?